traversal/executor/pattern_executor.rs

```diff
@@ -399,16 +439,16 @@ impl SortedExecutor {
         loop {
             let mut failed = false;
             let mut retry = false;
-            for i in 0..self.intersection_iterators.len() {
+            for i in 0..self.iterators.len() {
                 if i == current_max_index {
                     continue;
                 }
 
                 let (containing_i, containing_max, i_index, max_index) = if current_max_index > i {
-                    let (containing_i, containing_max) = self.intersection_iterators.split_at_mut(current_max_index);
+                    let (containing_i, containing_max) = self.iterators.split_at_mut(current_max_index);
                     (containing_i, containing_max, i, 0)
                 } else {
-                    let (containing_max, containing_i) = self.intersection_iterators.split_at_mut(i);
+                    let (containing_max, containing_i) = self.iterators.split_at_mut(i);
                     (containing_i, containing_max, 0, current_max_index)
                 };
```

Comment:
What is this for? Getting two mutable references?

Reply:
Yes